### ISSUE TITLE
Enhance writeLocalFile to determine output directory

### ISSUE THREAD

**Issue by TonyGravagno** [araera111/mysql-to-zod#174]
In [this recent rewrite](https://github.com/TonyGravagno/mysql-to-zod/commit/0cb6418a7621166ee7446009afd09147d6305ea2) of outputToFile.ts, other functions calculate the outDir and pass it to writeLocalFile.

With [this new trivial enhancement](https://github.com/TonyGravagno/mysql-to-zod/tree/tg-refine-local-write-1), other functions can, but are not obligated to be responsible for the outDir. Now they just need to pass the information they have, and writeLocalFile will determine the outDir.

```ts
export const writeLocalFile = (outputTo: string | OptionOutput | undefined, ... ) {
  const outDir = typeof outputTo === "string" && outputTo.length > 0
      ? outputTo
      : (typeof outputTo === "object" && outputTo?.outDir) || outputDefaults.outDir;
```

- If a non-empty string is passed, that is used for the outDir.
- If an `OptionOutput` object is used, outDir is extracted from the configuration
- If neither of the above are true, the default folder is used.

This "convenience" feature will be used in the next update.  
Rather than merge this and then merge the next one, you might just want to wait for the next one which will be posted very soon.

**Comment by TonyGravagno** [comment]
Change : I merged this in my fork with the current 2.3.0 develop branch. This is now wrapped in https://github.com/TonyGravagno/mysql-to-zod/tree/tg-save-ast-2

**Comment by araera111** [comment]
thanks merged.
